lib/Client.php

```diff
@@ -134,7 +132,7 @@ public function transaction() {
      */
     public function close() {
         /** @var Promise $promise */
-        $promise = all(promises($this->promisors));
+        $promise = all($this->promisors);
```

Comment:
The point is that Deferred implementing Promise in production is a mere optimization to save an extra object creation (and Promises are being created quite often, so…).

You always should assume it not implementing Promise.